internal/operator/cluster/upgrade.go

```diff
@@ -838,3 +846,90 @@ func publishUpgradeClusterFailureEvent(eventHeader events.EventHeader, clusterna
 		log.Errorf("error publishing event. Error: %s", err.Error())
 	}
 }
+
+// updateClusterConfig updates PG configuration for cluster via its Distributed Configuration Store
+// (DCS) according to the key/value pairs defined in the pgConfig map, specifically by updating
+// the <clusterName>-pgha-config ConfigMap.  The configuration settings specified are
+// applied to the entire cluster via the DCS configuration included within this the
+// <clusterName>-pgha-config ConfigMap.
+func updateClusterConfig(clientset kubeapi.Interface, pgcluster *crv1.Pgcluster, namespace string) error {
+
+	// first, define the names for the two main sections of the <clustername>-pgha-config configmap
+
+	// <clustername>-dcs-config
+	dcsConfigName := fmt.Sprintf(pgoconfig.PGHADCSConfigName, pgcluster.Name)
+	// <clustername>-local-config
+	localConfigName := fmt.Sprintf(pgoconfig.PGHALocalConfigName, pgcluster.Name)
+
+	// next, get the <clustername>-pgha-config configmap
+	clusterConfig, err := clientset.CoreV1().ConfigMaps(namespace).Get(context.TODO(), fmt.Sprintf("%s-pgha-config", pgcluster.Name), metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// prepare DCS config struct
+	dcsConf := &pgoconfig.DCSConfig{}
+	if err = yaml.Unmarshal([]byte(clusterConfig.Data[dcsConfigName]), dcsConf); err != nil {
+		return err
+	}
+
+	// prepare LocalDB config struct and nested structs
+	localDBConf := &pgoconfig.LocalDBConfig{}
+	callbacks := &pgoconfig.Callbacks{}
+	pgbackrest := &pgoconfig.CreateReplicaMethod{}
+	pgbackrestStandby := &pgoconfig.CreateReplicaMethod{}
+	if err = yaml.Unmarshal([]byte(clusterConfig.Data[localConfigName]), localDBConf); err != nil {
+		return err
+	}
+
+	// set the updated path values for both DCS and LocalDB configs
+	// as of version 4.6, the /crunchyadm directory no longer exists (previously set as a unix socket directory)
+	// and the /opt/cpm... directories are now set under /opt/crunchy
+	dcsConf.PostgreSQL.Parameters["unix_socket_directories"] = "/tmp"
+	dcsConf.PostgreSQL.Parameters["archive_command"] = `source /opt/crunchy/bin/postgres-ha/pgbackrest/pgbackrest-set-env.sh && pgbackrest archive-push "%p"`
+	dcsConf.PostgreSQL.RecoveryConf["restore_command"] = `source /opt/crunchy/bin/postgres-ha/pgbackrest/pgbackrest-set-env.sh && pgbackrest archive-get %f "%p"`
+
+	callbacks.OnRoleChange = "/opt/crunchy/bin/postgres-ha/callbacks/pgha-on-role-change.sh"
+	localDBConf.PostgreSQL.Callbacks = callbacks
+
+	pgbackrest.Command = "/opt/crunchy/bin/postgres-ha/pgbackrest/pgbackrest-create-replica.sh replica"
+	localDBConf.PostgreSQL.PGBackRest = pgbackrest
+
+	pgbackrestStandby.Command = "/opt/crunchy/bin/postgres-ha/pgbackrest/pgbackrest-create-replica.sh standby"
+	localDBConf.PostgreSQL.PGBackRestStandby = pgbackrestStandby
```

Comment:
I did some testing, and values that should've been kept were being overwritten. Thanks for the catch!

To correct this, I've updated this section to use the `localDBConf` struct, which received the unmarshalled data from the existing CM, to set the new values directly. This allows the existing struct values to stay in place rather than being lost during the update.

Reply:
WFM 👍